src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -40,416 +44,554 @@
+                pw.write("GET http://" + thirdpartyHost + ":" + thirdpartyPort + "/ HTTP/1.1\n");
+                pw.write("Host: " + thirdpartyHost + "\n\n");
+                pw.flush();
+
+                // read the response via a 4k buffer
+                bos = new ByteArrayOutputStream();
+                byte[] buffer = new byte[4096];
+                int bytesRead = is.read(buffer);
+                int totalBytesRead = 0;
+                while (bytesRead > -1) {
+                    totalBytesRead += bytesRead;
+                    bos.write(buffer, 0, bytesRead);
+                    bytesRead = is.read(buffer);
+                }
+                String response = new String(bos.toByteArray());
+                if (log.isDebugEnabled())
+                    log.debug("Response is " + totalBytesRead + " bytes: \n" + response);
+
+                Matcher m = thirdPartyContentPattern.matcher(response);
+                if (m.matches()) {
+                    log.debug("Response matches expected third party pattern!");
+                    bingo(Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM, Constant.messages.getString("ascanbeta.insecurehttpmethod.detailed.name", HttpRequestHeader.CONNECT),
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.desc", HttpRequestHeader.CONNECT), null,
+                            null, // parameter being attacked: none.
+                            "", // attack
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.connect.exploitable.extrainfo", thirdpartyHost),
+                            Constant.messages.getString("ascanbeta.insecurehttpmethod.soln"), response, // evidence,
+                            this.getBaseMsg());
+                    return;
+                } else {
+                    log.debug("Response does *not* match expected third party pattern");
+                }
+                return;
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
+                    log.debug("The status line returned: " + statusLine);
+                }
+                return;
+            }
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
+            }
+        }finally{
+            try{is.close();}catch(Exception e){}
+            try{os.close();}catch(Exception e){}
+            try{bos.close();}catch(Exception e){}
+            try{socket.close();}catch(Exception e){}
+        }
+    }
+
+    private final void testHttpMethod(String httpMethod) throws Exception {
+        //TODO:chose a body that exploits a vulnerability
```

Comment:
Not sure we really want to exploit something in this scanner, this is just for information of potentially insecure methods.